packages/core/src/components/graphs/scatter.ts

```diff
@@ -54,19 +54,19 @@ export class Scatter extends Component {
 		dotsEnter.merge(dots)
 			.raise()
 			.classed("dot", true)
-			.classed("filled", filled)
-			.classed("unfilled", !filled)
+			.classed("filled", d => this.model.getIsFilled(d.datasetLabel, d.label, d.value, d, filled))
+			.classed("unfilled", d => !this.model.getIsFilled(d.datasetLabel, d.label, d.value, d, filled))
 			.attr("cx", (d, i) => this.services.axes.getXValue(d, i))
 			.transition(this.services.transitions.getTransition("scatter-update-enter", animate))
 			.attr("cy", (d, i) => this.services.axes.getYValue(d, i))
 			.attr("r", options.points.radius)
 			.attr("fill", d => {
-				if (filled) {
-					return this.model.getFillScale()[d.datasetLabel](d.label) as any;
+				if (this.model.getIsFilled(d.datasetLabel, d.label, d.value, d, filled)) {
```

Comment:
again we don't need to re-call `this.model.getIsFilled` every time

Reply:
see above